linkerd/core/src/main/scala/io/buoyant/linkerd/Router.scala

```diff
@@ -202,10 +213,16 @@ case class RetriesConfig(
 ) {
 
   @JsonIgnore
-  def mk: Retries.Budget = Retries.Budget(
-    budget.map(_.mk).getOrElse(RetryBudget()),
-    backoff.map(_.mk).getOrElse(Backoff.const(Duration.Zero))
-  )
+  def mkBackoff: Option[ClassifiedRetries.Backoffs] =
+    backoff.map(_.mk).map(ClassifiedRetries.Backoffs(_))
+
+  // We use an empty backoff for Retries.Budget, since this informs
+  // _requeue_ delay. Requeues are explicitly for Nacks and
+  // non-application-level failures, and so we want to reenqueue
+  // these as quickly as possible.
+  @JsonIgnore
+  def mkBudget: Option[Retries.Budget] =
+    budget.map { b => Retries.Budget(b.mk, Backoff.const(Duration.Zero)) }
```

Comment:
TIOLI could use `Retries.Budget.emptyBackoffSchedule` instead of `Back.const(Duration.Zero)`